src/rapidhtml/tags.py

```diff
@@ -121,11 +122,53 @@ def __init__(
             )
 
         self.__closing_tag = f"</{self.tag}>" if not self.__self_closing else "/>"
+        self.__uuid: str = (
+            uuid4().hex
+        )  # A unique ID used to detect cyclical references of tags
 
     def __init_subclass__(cls, self_closing: bool = False, **kwargs) -> None:
         super().__init_subclass__(**kwargs)
         cls.__self_closing = self_closing
 
+    @property
+    def _uuid(self) -> str:
+        return self.__uuid
+
+    def validate_cyclical_references(
```

Comment:
Should we prepend with `_`?
```suggestion
    def _validate_cyclical_references(
```